castervoice/apps/ide/ide.py

```diff
@@ -0,0 +1,94 @@
+#
+# __author__ = "lexxish"
+#
+from castervoice.apps.shared.directions import FORWARD, RIGHT, BACK, LEFT, UP, DOWN
+
+method = "(meth|method)"
+
+# general
+EXPAND_SELECTION = "expand [selection] [<n>]"
+SHRINK_SELECTION = "shrink [selection] [<n>]"
+SMART_AUTO_COMPLETE = "(skraken|smart kraken)"
+GENERATE_CODE = "jen code"
+QUICK_FIX = "quickfix"
+FORMAT_ALL_CODE = "format [code]"
+BUILD_PROJECT = "build"
+RUN_PROJECT = "run"
+DEBUG_PROJECT = "debug"
+BUILD_AND_RUN_PROJECT = "build and run"
+DEBUG_CURRENT_FILE = "debug file"
+RUN_CURRENT_FILE = "run file"
+NEXT_ERROR = "(%s error|error %s)" % (FORWARD, RIGHT)
+PREVIOUS_ERROR = "(%s error|error %s)" % (BACK, LEFT)
+REDO = "redo [<n>]"
+SHOW_SETTINGS = "[show] settings"
+RENAME_CURRENT_FILE = "file rename | rename file"
+
+# window navigation
+NEXT_TAB = "%s tab [<n>]|tab %s [<n>]" % (FORWARD, RIGHT)
+PREVIOUS_TAB = "%s tab [<n>]|tab %s [<n>]" % (BACK, LEFT)
+CLOSE_TAB_N_TIMES = "close tab [<n>]|tab close [<n>]"
+GO_TO_EDITOR = "go [to] editor"
```

Comment:
I think we should keep "go" for text navigation and use "focus" for moving between areas of the IDE.

Reply:
"go back" "go forward" in the browser/file dialogue are probably fine.

Reply:
Yes "go" might not be the best here. Do you prefer "focus" over "show"

Reply:
focus is better. Many IDEs have different panes that are all visible (i.e. the editor may already be "shown" but not "focused"